zio-http-example/src/main/scala/example/StaticServer.scala

```diff
@@ -3,47 +3,53 @@ package example
 import zio.ZIOAppDefault
 
 import zio.http._
+import zio.http.codec.PathCodec.trailing
 import zio.http.html._
 
 object StaticServer extends ZIOAppDefault {
 
   // A simple app to serve static resource files from a local directory.
-  val app = Http
-    .collectHandler[Request] { case Method.GET -> "" /: "static" /: path =>
+  val app = Routes(
+    Method.GET / "static" / trailing -> handler {
+      val extractPath    = Handler.param[(Path, Request)](_._1)
```

Comment:
Idea: we could have a version of `handler` where the function's parameters are not the values, but these extractor handlers

Reply:
You mean something like, `handlerParam { (param1, param2) => ... }`, where `param1`, etc., are `Handler` values that extract?

Reply:
yes

Reply:
I think you would have to specify a lot of types, because the in and outs of the handlers would need to be known, so it would be something like: `handlerParam[(String, Int, Path), String, Int, Path]` or maybe you could do `handlerParam[(String, Int, Path)](_._1, _._2, _._3) { (param1, param2, param3) => ...}`.

Honestly I do not like this example a lot. I rewrote it in the fastest way possible, not necessarily the best way.

The difficulty is that it is a handler of handler situation: where the inner handlers want to use just the request, and the outer handlers want to use more information than the request (request + params).

It would be nice to find a good way to handle these scenarios.